### PR TITLE
Clang CMake job

### PR DESCRIPTION
**Pull request by johnmcfarlane**
Breakdown of supporting changes from #143.

### CODE REVIEW

**Comment by johnmcfarlane** [comment]
@ckormanyos this one isn't a draft PR. It breaks out the non-Clang-Tidy-related changes:

* adds a Clang job,
* rearranges the .yml file (the new Clang job is likely to take much longer than the others and putting a job first means it will start sooner and perform checks sooner),
* uses a Docker image, [_johnmcfarlane/cnl_ci:gcc-11_](https://hub.docker.com/layers/johnmcfarlane/cnl_ci/gcc-11/images/sha256-7ae8e18e09c1a2429bc1583fbedbae2ae08722a747a9a5502f3523437271e2fb?context=explore), which already contains Boost, so no need to clone and build it in the pipeline.

You may be interested in using this change on its own, but it's mostly in support of the Clang-Tidy PR I'm working on and unclutters that branch. LMK your thought.

File: .github/workflows/wide_integer.yml [hunk 1]
```diff
@@ -7,6 +7,76 @@ on:
   schedule:
     - cron: '0 2 * * *' # run at 2 AM UTC
 jobs:
+  cmake-linux:
+    strategy:
+      fail-fast: false
+      matrix:
+        compiler: [ g++, clang++ ]
```
Comment (johnmcfarlane) [inline .github/workflows/wide_integer.yml:14]: @ckormanyos obviously, to merge this change, you'd need to remove two jobs and add three to the list of required PR jobs.

File: .github/workflows/wide_integer.yml [hunk 2]
```diff
@@ -7,6 +7,76 @@ on:
...
+        include:
+          - compiler: g++
+            container: johnmcfarlane/cnl_ci:gcc-11
```
Comment (johnmcfarlane) [inline .github/workflows/wide_integer.yml:17]: I have many more of these images. If you were interested in speeding up jobs by downloading ready-baked environments with compilers, Boost, and other necessities, we could consider using these images in the other jobs or uploading your own to Docker Hub.

File: .github/workflows/wide_integer.yml [hunk 3]
```diff
@@ -7,6 +7,76 @@ on:
...
+        run: mkdir $GITHUB_WORKSPACE/build
+
+      - name: build
+        working-directory: build
+        run: |
+          cmake \
+              -DCMAKE_BUILD_TYPE=Release \
+              -DCMAKE_CXX_STANDARD=20 \
+              -DCMAKE_TOOLCHAIN_FILE=$GITHUB_WORKSPACE/.github/toolchains/gcc.cmake \
+              -GNinja \
```
Comment (johnmcfarlane) [inline .github/workflows/wide_integer.yml:36]: @ckormanyos `ninja` is a faster build tool than `make`. It also builds objects in parallel by default so build time for this job should be reduced quite a bit.

File: .github/workflows/wide_integer.yml (hunk 1 above)
Comment (ckormanyos) [reply .github/workflows/wide_integer.yml:14]: > to merge this change, you'd need to remove two jobs and add three to the list of required PR jobs

Done. Thank you @johnmcfarlane.

**Pull request merged**